intersect.go

```diff
@@ -176,6 +176,18 @@ func Without[T comparable, Slice ~[]T](collection Slice, exclude ...T) Slice {
 	return result
 }
 
+// WithoutBy filters a slice by excluding elements whose extracted keys match any in the exclude list.
+// It returns a new slice containing only the elements whose keys are not in the exclude list.
+func WithoutBy[T any, K comparable](collection []T, extract func(item T) K, exclude ...K) []T {
+	result := make([]T, 0, len(collection))
+	for _, item := range collection {
+		if !Contains(exclude, extract(item)) {
```

Comment:
I'd say it's worth using sets here for better performance
```go
func WithoutBy[T any, K comparable](collection []T, extract func(item T) K, exclude ...K) []T {
	whitelist := make(map[K]struct{}, len(collection))

	for _, item := range collection {
		whitelist[extract(item)] = struct{}{}
	}

	for _, k := range exclude {
		delete(whitelist, k)
	}

	result := make([]T, 0, len(whitelist))
	for _, item := range collection {
		if _, ok := whitelist[extract(item)]; ok {
			result = append(result, item)
		}
	}

	return result
}
```

Might as well refactor existing function `Without`

Reply:
And maybe It would be good to simply call `WithoutBy` in `Without`, most likey compiler will optimize such a call (could check in godbolt)
```go
func Without[T comparable](collection []T, exclude ...T) []T {
	return WithoutBy(collection, func(item T) T { return item }, exclude...)
}
```

Reply:
I have refactored WithoutBy function. I used `blacklist` to replace `whitelist` for code readability. It's clear now. Please review it again.